api/src/main/scala/ai/chronon/api/Extensions.scala

```diff
@@ -634,11 +636,63 @@ object Extensions {
     def sanitize: String = Option(string).map(_.replaceAll("[^a-zA-Z0-9_]", "_")).orNull
 
     def cleanSpec: String = string.split("/").head
+  }
 
+  implicit class filePathOps(filePath: String) {
+    @transient lazy val logger = LoggerFactory.getLogger(getClass)
+    // process chronon configs only. others will be ignored
+    // todo: add metrics
+
+    private def loadJsonToConf[T <: TBase[_, _]: Manifest: ClassTag](file: String): Option[T] = {
+      try {
+        val configConf = ThriftJsonCodec.fromJsonFile[T](file, check = true)
+        Some(configConf)
+      } catch {
+        case e: Throwable =>
+          logger.error(s"Failed to parse compiled Chronon config file: $file, \nerror=${e.getMessage}")
+          None
+      }
+    }
+    private def loadJson[T <: TBase[_, _]: Manifest: ClassTag](file: String): Option[String] = {
+      try {
+        val configConf = loadJsonToConf[T](file).get
+        Some(ThriftJsonCodec.toJsonStr(configConf))
+      } catch {
+        case e: Throwable =>
+          logger.error(s"Failed to parse compiled Chronon config file: $file, \nerror=${e.getMessage}")
+          None
+      }
+    }
     // derive a feature name key from path to file
+    // capture <conf_type>/<team>/<conf_name> as key e.g joins/team/team.example_join.v1
     def confPathToKey: String = {
-      // capture <conf_type>/<team>/<conf_name> as key e.g joins/team/team.example_join.v1
-      string.split("/").takeRight(3).mkString("/")
+      filePath.split("/").takeRight(3).mkString("/")
+    }
+
+    // derive a feature team key from config, e.g: joins/team
+    def confPathToTeamKey: Option[String] = {
```

Comment:
So we should make it so that we only parse the file once and run a bunch of extractors for the conf type.

So we need to maintain a list of (key, value) extractors for groupBys, joins and stagingQueries.

We simply parse once, run all the extractors per file, print it to console and write to metastore.

Reply:
Hi @nikhilsimha this function is only called in the defined extract function in metadata end point. The metadata dir walker will go through the files like you mentioned here. I explained the logic below in metadata DirWalker.